appdaemon/state.py

```diff
@@ -101,17 +101,16 @@ def terminate(self):
             # In the case of a quick_start parameter,
             # start the clock immediately if the device is already in the new state
             #
-            if "immediate" in kwargs and kwargs["immediate"] is True:
-                __duration = 0 # run it immediately
+            if kwargs.get("immediate"):
```

Comment:
@basnijholt this area will lead to a break in code, since immediate is not always supplied. so should be kwargs.get("immediate", False). But the issue with even using this method, is the user could supply any string
and it will still go thru. so should be 
```
if kwargs.get("immediate", False) is True
```

Reply:
In the docs it says immediate is a bool and by default `get` returns `None`, and `bool(None) == True`, so this is exactly the same.

One should never do `if attr is True` **unless** the variable can be different types, like integers, but this is not the case here.

Reply:
@basnijholt I think you mean `bool(None) != True`, right?

What we want here is if a user supplied `True` for immediate, then the immediate happens. If the user supplies nothing, or anything other than `True` (including `"yes"` and `"True"`) then the immediate should not happen.

So, I believe the most concise way to write this is a combination of the @basnijholt  and @Odianosen25 code:
```
if kwargs.get("immediate") is True:
```

Reply:
Yes, I meant `bool(None) != True`.

If that behaviour is required then the `... is True` is indeed the correct solution.

But wouldn't it be a better user-experience if we explicitly check the input types? Because if someone writes `"on"`, a user might expect that the code works (as no error is raised.)

Reply:
It would be a better user-experience, absolutely, if input types were checked and an error raised if the input does not validate. I was merely aiming for "same functionality as before" with this.

@acockburn Agrees that input validation is an eventual goal and has mentioned Voluptuous a few times as a method of getting there. 

For many Python APIs, Voluptuous is overkill on a simple method call because Python programmers can generally work this out on their own. But, in the case of AppDaemon, we are encouraging VERY NEW programmers to write Python. Anything we can do to provide a clear error is helpful to them. And a user writing immediate="True" instead of immediate=True is a fairly common mistake.

Reply:
What about validating the types based on type annotations and using the following decorator:
```python
from functools import wraps
from inspect import getfullargspec
from typing import get_type_hints

def validate_input(obj, **kwargs):
    for name, type_ in get_type_hints(obj).items():
        if name == 'return':
            continue

        if not isinstance(kwargs[name], type_):
            raise TypeError(
                f'Argument {name} is not of type {type}'
            )


def type_check(f):
    @wraps(f)
    def wrapped_decorator(*args, **kwargs):
        # convert *args into **kwargs
        func_args = getfullargspec(f)[0]
        kwargs.update(dict(zip(func_args, args)))

        validate_input(f, **kwargs)
        return f(**kwargs)

    return wrapped_decorator
```

Reply:
Oh, in this case, that would quite work actually, as the function uses `kwargs`.